File: app/src/main/java/com/cyl/musiclake/api/baidu/BaiduSongInfo.java
package com.cyl.musiclake.api.baidu;

import java.util.List;

/**
 * 作者：yonglong on 2016/9/9 04:08
 * 邮箱：643872807@qq.com
 * 版本：2.5
 */
public class BaiduSongInfo {

    /**
     * errorCode : 22000
     * data : {"xcode":"ad6e8c055ff4dfd9261032699fa1be7e","songList":[{"queryId":"569080829","songId":569080829,"songName":"无问西东","artistId":"45561","artistName":"王菲","albumId":569080827,"albumName":"无问西东","songPicSmall":"http://qukufile2.qianqian.com/data2/pic/4865939a77b87edc79789df87b6f22d8/569080825/569080825.png@s_1,w_90,h_90","songPicBig":"http://qukufile2.qianqian.com/data2/pic/4865939a77b87edc79789df87b6f22d8/569080825/569080825.png@s_1,w_150,h_150","songPicRadio":"http://qukufile2.qianqian.com/data2/pic/4865939a77b87edc79789df87b6f22d8/569080825/569080825.png@s_1,w_300,h_300","lrcLink":"http://qukufile2.qianqian.com/data2/lrc/ef80d282b94f37e92bee6e5b9b417124/569080826/569080826.lrc","version":"影视原声","copyType":0,"time":290,"linkCode":22000,"songLink":"http://zhangmenshiting.qianqian.com/data2/music/b008dff3c9df62831582ddea13f64921/569080852/569080829104400128.mp3?xcode=ad6e8c055ff4dfd92bb21879f3eb947d","showLink":"http://zhangmenshiting.qianqian.com/data2/music/b008dff3c9df62831582ddea13f64921/569080852/569080829104400128.mp3?xcode=ad6e8c055ff4dfd92bb21879f3eb947d","format":"mp3","rate":128,"size":4638118,"relateStatus":"0","resourceType":"0","source":"web"}]}
     */

    private int errorCode;
    private DataBean data;

    public int getErrorCode() {
        return errorCode;
    }

    public void setErrorCode(int errorCode) {
        this.errorCode = errorCode;
    }

    public DataBean getData() {
        return data;
    }

    public void setData(DataBean data) {
        this.data = data;
    }

    public static class DataBean {
        /**
         * xcode : ad6e8c055ff4dfd9261032699fa1be7e
         * songList : [{"queryId":"569080829","songId":569080829,"songName":"无问西东","artistId":"45561","artistName":"王菲","albumId":569080827,"albumName":"无问西东","songPicSmall":"http://qukufile2.qianqian.com/data2/pic/4865939a77b87edc79789df87b6f22d8/569080825/569080825.png@s_1,w_90,h_90","songPicBig":"http://qukufile2.qianqian.com/data2/pic/4865939a77b87edc79789df87b6f22d8/569080825/569080825.png@s_1,w_150,h_150","songPicRadio":"http://qukufile2.qianqian.com/data2/pic/4865939a77b87edc79789df87b6f22d8/569080825/569080825.png@s_1,w_300,h_300","lrcLink":"http://qukufile2.qianqian.com/data2/lrc/ef80d282b94f37e92bee6e5b9b417124/569080826/569080826.lrc","version":"影视原声","copyType":0,"time":290,"linkCode":22000,"songLink":"http://zhangmenshiting.qianqian.com/data2/music/b008dff3c9df62831582ddea13f64921/569080852/569080829104400128.mp3?xcode=ad6e8c055ff4dfd92bb21879f3eb947d","showLink":"http://zhangmenshiting.qianqian.com/data2/music/b008dff3c9df62831582ddea13f64921/569080852/569080829104400128.mp3?xcode=ad6e8c055ff4dfd92bb21879f3eb947d","format":"mp3","rate":128,"size":4638118,"relateStatus":"0","resourceType":"0","source":"web"}]
         */

        private String xcode;
        private List<SongListBean> songList;

        public String getXcode() {
            return xcode;
        }

        public void setXcode(String xcode) {
            this.xcode = xcode;
        }

        public List<SongListBean> getSongList() {
            return songList;
        }

        public void setSongList(List<SongListBean> songList) {
            this.songList = songList;
        }

        public static class SongListBean {
            /**
             * queryId : 569080829
             * songId : 569080829
             * songName : 无问西东
             * artistId : 45561
             * artistName : 王菲
             * albumId : 569080827
             * albumName : 无问西东
             * songPicSmall : http://qukufile2.qianqian.com/data2/pic/4865939a77b87edc79789df87b6f22d8/569080825/569080825.png@s_1,w_90,h_90
             * songPicBig : http://qukufile2.qianqian.com/data2/pic/4865939a77b87edc79789df87b6f22d8/569080825/569080825.png@s_1,w_150,h_150
             * songPicRadio : http://qukufile2.qianqian.com/data2/pic/4865939a77b87edc79789df87b6f22d8/569080825/569080825.png@s_1,w_300,h_300
             * lrcLink : http://qukufile2.qianqian.com/data2/lrc/ef80d282b94f37e92bee6e5b9b417124/569080826/569080826.lrc
             * version : 影视原声
             * copyType : 0
             * time : 290
             * linkCode : 22000
             * songLink : http://zhangmenshiting.qianqian.com/data2/music/b008dff3c9df62831582ddea13f64921/569080852/569080829104400128.mp3?xcode=ad6e8c055ff4dfd92bb21879f3eb947d
             * showLink : http://zhangmenshiting.qianqian.com/data2/music/b008dff3c9df62831582ddea13f64921/569080852/569080829104400128.mp3?xcode=ad6e8c055ff4dfd92bb21879f3eb947d
             * format : mp3
             * rate : 128
             * size : 4638118
             * relateStatus : 0
             * resourceType : 0
             * source : web
             */

            private String queryId;
            private String songId;
            private String songName;
            private String artistId;
            private String artistName;
            private long albumId;
            private String albumName;
            private String songPicSmall;
            private String songPicBig;
            private String songPicRadio;
            private String lrcLink;
            private String version;
            private int copyType;
            private int time;
            private int linkCode;
            private String songLink;
            private String showLink;
            private String format;
            private int rate;
            private long size;
            private String relateStatus;
            private String resourceType;
            private String source;

            public String getQueryId() {
                return queryId;
            }

            public void setQueryId(String queryId) {
                this.queryId = queryId;
            }

            public String getSongId() {
                return songId;
            }

            public void setSongId(String songId) {
                this.songId = songId;
            }

            public String getSongName() {
                return songName;
            }

            public void setSongName(String songName) {
                this.songName = songName;
            }

            public String getArtistId() {
                return artistId;
            }

            public void setArtistId(String artistId) {
                this.artistId = artistId;
            }

            public String getArtistName() {
                return artistName;
            }

            public void setArtistName(String artistName) {
                this.artistName = artistName;
            }

            public long getAlbumId() {
                return albumId;
            }

            public void setAlbumId(long albumId) {
                this.albumId = albumId;
            }

            public String getAlbumName() {
                return albumName;
            }

            public void setAlbumName(String albumName) {
                this.albumName = albumName;
            }

            public String getSongPicSmall() {
                return songPicSmall;
            }

            public void setSongPicSmall(String songPicSmall) {
                this.songPicSmall = songPicSmall;
            }

            public String getSongPicBig() {
                return songPicBig;
            }

            public void setSongPicBig(String songPicBig) {
                this.songPicBig = songPicBig;
            }

            public String getSongPicRadio() {
                return songPicRadio;
            }

            public void setSongPicRadio(String songPicRadio) {
                this.songPicRadio = songPicRadio;
            }

            public String getLrcLink() {
                return lrcLink;
            }

            public void setLrcLink(String lrcLink) {
                this.lrcLink = lrcLink;
            }

            public String getVersion() {
                return version;
            }

            public void setVersion(String version) {
                this.version = version;
            }

            public int getCopyType() {
                return copyType;
            }

            public void setCopyType(int copyType) {
                this.copyType = copyType;
            }

            public int getTime() {
                return time;
            }

            public void setTime(int time) {
                this.time = time;
            }

            public int getLinkCode() {
                return linkCode;
            }

            public void setLinkCode(int linkCode) {
                this.linkCode = linkCode;
            }

            public String getSongLink() {
                return songLink;
            }

            public void setSongLink(String songLink) {
                this.songLink = songLink;
            }

            public String getShowLink() {
                return showLink;
            }

            public void setShowLink(String showLink) {
                this.showLink = showLink;
            }

            public String getFormat() {
                return format;
            }

            public void setFormat(String format) {
                this.format = format;
            }

            public int getRate() {
                return rate;
            }

            public void setRate(int rate) {
                this.rate = rate;
            }

            public long getSize() {
                return size;
            }

            public void setSize(long size) {
                this.size = size;
            }

            public String getRelateStatus() {
                return relateStatus;
            }

            public void setRelateStatus(String relateStatus) {
                this.relateStatus = relateStatus;
            }

            public String getResourceType() {
                return resourceType;
            }

            public void setResourceType(String resourceType) {
                this.resourceType = resourceType;
            }

            public String getSource() {
                return source;
            }

            public void setSource(String source) {
                this.source = source;
            }
        }
    }
}

File: musicapi/src/main/java/com/cyl/musicapi/baidu/BaiduSongInfo.kt
package com.cyl.musicapi.baidu


import com.google.gson.annotations.SerializedName

//"songList": [{
//			"queryId": "265715650",
//			"songId": 265715650,
//			"songName": "\u5927\u9c7c",
//			"artistId": "164528737",
//			"artistName": "\u5468\u6df1",
//			"albumId": 265715651,
//			"albumName": "\u5927\u9c7c",
//			"songPicSmall": "http:\/\/qukufile2.qianqian.com\/data2\/pic\/0d359ec1be6f5365f92d4c83d3eeb022\/603758238\/603758238.jpg@s_2,w_90,h_90",
//			"songPicBig": "http:\/\/qukufile2.qianqian.com\/data2\/pic\/0d359ec1be6f5365f92d4c83d3eeb022\/603758238\/603758238.jpg@s_2,w_150,h_150",
//			"songPicRadio": "http:\/\/qukufile2.qianqian.com\/data2\/pic\/0d359ec1be6f5365f92d4c83d3eeb022\/603758238\/603758238.jpg@s_2,w_300,h_300",
//			"lrcLink": "http:\/\/qukufile2.qianqian.com\/data2\/lrc\/7a07a153b8a9a9d918bbee676b7d4118\/603149730\/603149730.lrc",
//			"version": "",
//			"copyType": 0,
//			"time": 313,
//			"linkCode": 22000,
//			"songLink": "http:\/\/zhangmenshiting.qianqian.com\/data2\/music\/a612909cdafecf20933bd2942c43421c\/596603939\/26571565036000128.mp3?xcode=ee75606752833ea05fa4084c7da1aa86",
//			"showLink": "http:\/\/zhangmenshiting.qianqian.com\/data2\/music\/a612909cdafecf20933bd2942c43421c\/596603939\/26571565036000128.mp3?xcode=ee75606752833ea05fa4084c7da1aa86",
//			"format": "mp3",
//			"rate": 128,
//			"size": 5021639,
//			"relateStatus": "0",
//			"resourceType": "0",
//			"source": "web"
//		}]
data class SongItem(@SerializedName("songName")
                        val songName: String = "",
                        @SerializedName("albumName")
                        val albumName: String = "",
                        @SerializedName("linkCode")
                        val linkCode: Int = 0,
                        @SerializedName("format")
                        val format: String = "",
                        @SerializedName("albumId")
                        val albumId: Int = 0,
                        @SerializedName("artistId")
                        val artistId: String = "",
                        @SerializedName("ting_uid")
                        val tingUid: String = "",
                        @SerializedName("source")
                        val source: String = "",
                        @SerializedName("songPicBig")
                        val songPicBig: String = "",
                        @SerializedName("version")
                        val version: String = "",
                        @SerializedName("queryId")
                        val queryId: String = "",
                        @SerializedName("songLink")
                        val songLink: String = "",
                        @SerializedName("size")
                        val size: Int = 0,
                        @SerializedName("rate")
                        val rate: Int = 0,
                        @SerializedName("lrcLink")
                        val lrcLink: String = "",
                        @SerializedName("copyType")
                        val copyType: Int = 0,
                        @SerializedName("artistName")
                        val artistName: String = "",
                        @SerializedName("time")
                        val time: Int = 0,
                        @SerializedName("relateStatus")
                        val relateStatus: String = "",
                        @SerializedName("songPicSmall")
                        val songPicSmall: String = "",
                        @SerializedName("songId")
                        val songId: Int = 0,
                        @SerializedName("songPicRadio")
                        val songPicRadio: String = "",
                        @SerializedName("showLink")
                        val showLink: String = "",
                        @SerializedName("resourceType")
                        val resourceType: String = "")


data class Data(@SerializedName("xcode")
                val xcode: String = "",
                @SerializedName("songList")
                val songList: List<SongItem>?)


data class BaiduSongInfo(@SerializedName("data")
                         val data: Data,
                         @SerializedName("errorCode")
                         val errorCode: Int = 0)



data class SongPlayRes(
    @SerializedName("bitrate")
    val bitrate: Bitrate,
    @SerializedName("error_code")
    val errorCode: Int,
    @SerializedName("songinfo")
    val songinfo: SongInfo
)

data class Bitrate(
    @SerializedName("file_bitrate")
    val fileBitrate: Int,
    @SerializedName("file_duration")
    val fileDuration: Int,
    @SerializedName("file_extension")
    val fileExtension: String,
    @SerializedName("file_link")
    val fileLink: String,
    @SerializedName("file_size")
    val fileSize: Int,
    @SerializedName("free")
    val free: Int,
    @SerializedName("hash")
    val hash: String,
    @SerializedName("show_link")
    val showLink: String,
    @SerializedName("song_file_id")
    val songFileId: Int
)

data class SongInfo(
    @SerializedName("album_id")
    val albumId: String,
    @SerializedName("album_no")
    val albumNo: String,
    @SerializedName("album_title")
    val albumTitle: String,
    @SerializedName("all_artist_id")
    val allArtistId: String,
    @SerializedName("all_artist_ting_uid")
    val allArtistTingUid: String,
    @SerializedName("all_rate")
    val allRate: String,
    @SerializedName("artist_id")
    val artistId: String,
    @SerializedName("author")
    val author: String,
    @SerializedName("bitrate_fee")
    val bitrateFee: String,
    @SerializedName("charge")
    val charge: Int,
    @SerializedName("copy_type")
    val copyType: String,
    @SerializedName("has_mv")
    val hasMv: Int,
    @SerializedName("has_mv_mobile")
    val hasMvMobile: Int,
    @SerializedName("havehigh")
    val havehigh: Int,
    @SerializedName("is_first_publish")
    val isFirstPublish: Int,
    @SerializedName("korean_bb_song")
    val koreanBbSong: String,
    @SerializedName("learn")
    val learn: Int,
    @SerializedName("lrclink")
    val lrclink: String,
    @SerializedName("piao_id")
    val piaoId: String,
    @SerializedName("pic_big")
    val picBig: String,
    @SerializedName("pic_huge")
    val picHuge: String,
    @SerializedName("pic_premium")
    val picPremium: String,
    @SerializedName("pic_radio")
    val picRadio: String,
    @SerializedName("pic_small")
    val picSmall: String,
    @SerializedName("play_type")
    val playType: Int,
    @SerializedName("relate_status")
    val relateStatus: String,
    @SerializedName("resource_type")
    val resourceType: String,
    @SerializedName("resource_type_ext")
    val resourceTypeExt: String,
    @SerializedName("si_proxycompany")
    val siProxycompany: String,
    @SerializedName("song_id")
    val songId: String,
    @SerializedName("song_source")
    val songSource: String,
    @SerializedName("special_type")
    val specialType: Int,
    @SerializedName("ting_uid")
    val tingUid: String,
    @SerializedName("title")
    val title: String,
    @SerializedName("toneid")
    val toneid: String
)